src/main/java/hudson/plugins/sauce_ondemand/SauceOnDemandBuildWrapper.java

```diff
@@ -549,16 +564,12 @@ private static String getCurrentHostName() {
     /**
      * @return the port to be used
      */
-    private int getPort() {
+    private int getPort(Map<String, String> envVars) {
         if (StringUtils.isNotBlank(seleniumPort) && !seleniumPort.equals("0")) {
             Matcher matcher = ENVIRONMENT_VARIABLE_PATTERN.matcher(seleniumPort);
             if (matcher.matches()) {
                 String variableName = matcher.group(1);
-                String value = System.getenv(variableName);
-                if (value == null) {
-                    value = "0";
-                }
-                return Integer.parseInt(value);
+                return MapUtils.getInteger(envVars, variableName, 0);
```

Comment:
I doubt there's such example, because...
* <code>String value = System.getenv(variableName)</code> takes the system variable on the server side (Jenkins master)
* <code>build.getEnvironment(listener)</code> should properly get System environment variables from the node, which executes the build

These two cases are orthogonal. The initial behavior is wrong IMHO, but users may depend on it.

My recommendation would be to... (pseudocode)

```java
String value = envVars.get(variableName);
if (value == null) {
  value = System.getenv(variableName);
}
if (value == null) {
  value = "0";
}
```

It would retain the original behavior if there is no Build environment variables (e.g. Node Global Variables, Build Parameters) specified